Document how to publish custom React Native version

bartolkaruza opened the pull request:
Add documentation page for #653 to the Android Building from Sources doc.

gengjiawen commented on docs/publishing.md:
````diff
@@ -0,0 +1,53 @@
+ This binary needs to become available in your project's `node_modules/react-native/android` folder or directly in your gradle dependency of your Android app. You can achieve this in one of two ways: Git dependency branch, Android binary dependency through Maven.
+
+ To build the .aar React Native library, you can follow the steps to [build from source](building-from-source.md) first to install all required tooling. Then to build the actual library, you can run the following in the root of your react-native checkout:
+ ```$bash
````
Can you consider make docker part TL, DR. And add links to docker website.

gengjiawen replied:
I want to show developers that make a fork is really simple.

gengjiawen commented on docs/publishing.md:
````diff
@@ -0,0 +1,53 @@
+Alternatively, if you don't want to 
+install the required toolchain for building from source, you can use a prebuilt docker image to create a react native binary;
+ ```
+ docker run --rm --name rn-build -v $PWD:/pwd -w /pwd gengjiawen/react-native /bin/sh -c "./gradlew installArchives"
````
Change `gengjiawen/react-native` to `reactnativecommunity/react-native-android`

gengjiawen commented on docs/publishing.md:
````diff
@@ -0,0 +1,53 @@
+```$bash
+# create .aar, then:
+sed -i '' "s|^/android/||g" .gitignore
+git add android
````
Maybe just `git add android --force`.

bartolkaruza commented:
@gengjiawen I've updated the PR for your comments. Can you clarify what you meant with add links to Docker website? I now added a single link to docker.com, but maybe you meant a specific page.

gengjiawen commented:
I think this will be good https://docs.docker.com/install/

bartolkaruza commented:
@gengjiawen could you check the docker command is correct? I don't have it installed on this machine (disk space :P), so I wasn't able to run it yet.

I think otherwise this PR is complete

gengjiawen commented:
LGTM. Great work.

bartolkaruza commented:
@grabbou you mentioned you guys maintain a lot of forks at Callstack, would you mind having a quick look at this one?

grabbou commented:
Will look at it tomorrow, thanks for CCing us.

grabbou commented:
@facebook-github-bot shipit

Our setup doesn't rely on Docker - I am taking @gengjiawen approval as a green light for this PR. Great job! And sorry for the lag, looks like Christmas break took longer than expected
